### PR TITLE
Doing selects

### PR DESCRIPTION
**Pull request by CG1235**

### CODE REVIEW

File: homework-selects/sql_questions.sql [hunk 1]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
+         JOIN EMPLOYEES M
+              ON E.MANAGER_ID = M.EMPLOYEE_ID ;
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT e.first_name AS EMPLOYEE_NAME, m.first_name as MANAGER_NAME
+FROM employees e
+         LEFT JOIN employees m ON e.manager_id = m.employee_id;
 -- 8. the details of employees who manage a department.
+SELECT *
+FROM EMPLOYEES
+         LEFT JOIN DEPARTMENTS USING (MANAGER_ID) WHERE MANAGER_ID IS NOT NULL AND DEPARTMENT_NAME IS NOT NULL ;
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:38]: Include in join conditions only the join logic itself, for filtering use where, it seems more readable, and the query might seem complex on multiple tables

File: homework-selects/sql_questions.sql [hunk 2]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
+         INNER JOIN departments USING (department_id)
+GROUP BY department_name;
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT  department_name, COUNT(employee_id) as Number_of_Employees, AVG(salary) as Average_Salary
+FROM employees
+         INNER JOIN departments USING (department_id)
+WHERE commission_pct IS NOT NULL
+GROUP BY department_name;
 --12. job title and average salary of employees.
+SELECT JOB_TITLE, AVG(SALARY)
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:57]: Give the names explicitly, it should show the table with AVG(COLUMN), we need Column_Name

File: homework-selects/sql_questions.sql [hunk 3]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
+FROM EMPLOYEES
+         NATURAL JOIN JOBS
+GROUP BY JOB_TITLE;
 --13. the country name, city, and number of those departments where at least 2 employees are working.
+SELECT COUNTRY_NAME,CITY, COUNT(DEPARTMENT_ID)
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:62]: Here put an alias department_nr for example, the name you choose, but it should be as in condition

File: homework-selects/sql_questions.sql [hunk 4]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
+       HAVING COUNT(department_id)>=2)
+GROUP BY country_name,city;
 --14. the employee ID, job name, number of days worked in for all those jobs in department 80.
+SELECT  employee_id, job_title, end_date - start_date as Number_of_Days
+FROM employees
+         JOIN jobs USING (job_id)
+         JOIN job_history USING (employee_id)
+WHERE job_history.department_id = 80;
 --15. the name ( first name and last name ) for those employees who gets more salary than the employee whose ID is 163.
+SELECT FIRST_NAME, LAST_NAME
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:79]: the name (...) means to show 1 column with te concatenation of fields, you should show the columns exactly as they are listed in the condition, check next conditions too

File: homework-selects/sql_questions.sql [hunk 5]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
+SELECT (e.first_name ||  ' ' || e.last_name) name, e.employee_id, e.salary
+FROM employees e
+         JOIN employees m ON e.manager_id = m.employee_id and m.first_name = 'Payam';
 --18. the department number, name ( first name and last name ), job and department name for all employees in the Finance department.
+SELECT DEPARTMENTS.DEPARTMENT_ID, FIRST_NAME, LAST_NAME, JOB_TITLE, DEPARTMENT_NAME
+FROM DEPARTMENTS, JOBS, EMPLOYEES
+WHERE DEPARTMENT_NAME = 'Finance';
 --19. all the information of an employee whose id is any of the number 134, 159 and 183.
+SELECT * FROM EMPLOYEES
+WHERE EMPLOYEE_ID = 134 OR EMPLOYEE_ID = 159 OR EMPLOYEE_ID =183;
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:96]: Choose a more appropriate operator from SQL, so it is 1 check

File: homework-selects/sql_questions.sql [hunk 6]
```diff
@@ -4,34 +4,164 @@ SELECT first_name, last_name, salary, job_title
...
 --20. all the information of the employees whose salary is within the range of smallest salary and 2500.
---21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT *
+FROM EMPLOYEES
+WHERE SALARY BETWEEN (SELECT MIN(SALARY)FROM EMPLOYEES) and 2500;
+ --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+ SELECT *
+FROM EMPLOYEES
+WHERE EMPLOYEE_ID BETWEEN 100 and 200 AND EMPLOYEE_ID != EMPLOYEES.DEPARTMENT_ID;
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:104]: This condition means to show the workers that do not work in a department with employees 100-200

**Review by ChislariSerafim — CHANGES_REQUESTED** [review]

File: homework-selects/sql_questions.sql (hunk 1 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: fixed

File: homework-selects/sql_questions.sql (hunk 2 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: fixed

File: homework-selects/sql_questions.sql (hunk 3 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: fixed

File: homework-selects/sql_questions.sql (hunk 4 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: done

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: done

File: homework-selects/sql_questions.sql [hunk 7]
```diff
@@ -4,34 +4,166 @@ SELECT first_name, last_name, salary, job_title
...
+WHERE employee_id IN (134, 159, 183);
 --20. all the information of the employees whose salary is within the range of smallest salary and 2500.
---21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT *
+FROM EMPLOYEES
+WHERE SALARY BETWEEN (SELECT MIN(SALARY)FROM EMPLOYEES) and 2500;
+ --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT *
+FROM employees
+WHERE employee_id NOT BETWEEN 100 AND 200;
```
Comment (ChislariSerafim) [inline homework-selects/sql_questions.sql:106]: You should exclude the employees, from departments where exmployess with id 100 to 200 work, it should be an additional inner select

**Review by ChislariSerafim — APPROVED** [review]

File: homework-selects/sql_questions.sql (hunk 7 above)
Comment (CG1235) [reply homework-selects/sql_questions.sql]: fixed

**Review by ChislariSerafim — APPROVED** [review]

**Review by ChislariSerafim — APPROVED** [review]

**Pull request merged**